### PR TITLE
docs - script and docs for list of releases

### PR DESCRIPTION
**Pull request by jeff-bruemmer**
PR includes:

- A page that list all of our releases, with links to the relevant GitHub release page. This list is useful for people who have been on an older Metabase version for a while, and want to upgrade to the latest version. 
- A babashka script to generate this page.

@nemanjaglumac ideally we incorporate running this script during/after the release process, but figured I should get it working as a standalone script first. It's pretty basic at the moment, but we could get fancier with it later on (e.g., break the list up by major versions and link to the relevant release announcement post and docs). I'll need to figure out how to add tests to the bb script.

The script requires babashka to run, as well as GitHub's `gh` CLI.

@nllho Anything we should add to the list of releases page? I still need to add links to the page from various places.

- - -
This change is [Reviewable](https://reviewable.io/reviews/metabase/metabase/30403)

### CODE REVIEW

**Comment by nllho** [comment]
@jeff-bruemmer for the list of releases page, maybe some quick links/notes on Upgrading and Downgrading?

- https://www.metabase.com/docs/latest/installation-and-operation/upgrading-metabase
- https://www.metabase.com/docs/latest/installation-and-operation/upgrading-metabase#rolling-back-an-upgrade

---

I also remember reading that you can only upgrade / downgrade one version at a time. So if you're on 38, you'd have to repeat the upgrade steps for versions 39, 40, ... until you get to 46 / current version.

Might be relevant to people who are looking for a list of versions (rather than searching for the upgrade docs themselves). 

@Tony-metabase I think I read this in one of your tickets if you could confirm :)

**Comment by jeff-bruemmer** [comment]
@nllho I added the supporting text, can you take a look?

@nemanjaglumac Can you try and run the script? See the README for setup.

**Comment by nemanjaglumac** [comment]
> I seem to remember reading that you can only upgrade / downgrade one version at a time. So if you're on 38, you'd have to repeat the upgrade steps for versions 39, 40, ... until you get to 46 / current version.

> Might be relevant to people who are looking for a list of versions (rather than searching for the upgrade docs themselves).

> @Tony-metabase I think I read this in one of your tickets? Or maybe @nemanjaglumac can confirm too :)

I don't remember a hard limitation like that one.
The bigger the gap, the more likely something will break. But I don't think anything is stopping anyone from trying an upgrade. They just have to have a backup first.

**Comment by Tony-metabase** [comment]
Correct, the only reason I recommend 39, 40, ... until you get to 46 / current version is so that if something happens in between I would know which version was the problem

File: bin/release-list/README.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+From this directory, run:
+
+```
+bb test:bb
+```
+
+
+## Editing the page
+
+Edit the file: [./resources/releases.md](./resources/releases.md).
```
Comment (nemanjaglumac) [inline bin/release-list/README.md:31]:
```suggestion
## Editing the page

Edit the template: [./resources/releases.md](./resources/releases.md).
```

Not sure if this change makes sense to you, but it's more explicit. I'd maybe even call this page `releases-template.md` even.

File: bin/release-list/README.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
+# Release list
+
+Builds a page that lists all Metabase releases for docs: `docs/releases.md`.
+
+## Requirements
+
+This script requires:
+
+- [Babashka](https://github.com/babashka/babashka)
+- [GitHub's CLI (gh)](https://cli.github.com/)
```
Comment (nemanjaglumac) [inline bin/release-list/README.md:10]:
```suggestion
- [GitHub's CLI (gh)](https://cli.github.com)
```

Nit

File: bin/release-list/test/release_list/main_test.clj [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+           {:edition 0
+            :major 45
+            :point 3
+            :hotfix 0}))))
+
+(def releases "
+Metabase v0.46.1                                                                                      Latest  v0.46.1     about 21 days ago
+Metabase® Enterprise Edition™ v1.46.1                                                                         v1.46.1     about 21 days ago
+Metabase v0.45.3.1                                                                                            v0.45.3.1   about 29 days ago
+Metabase® Enterprise Edition™ v1.45.3.1                                                                       v1.45.3.1   about 29 days ago")
```
Comment (nemanjaglumac) [inline bin/release-list/test/release_list/main_test.clj:37]: Whitespace seems to be off here 👀

**Review by nemanjaglumac — APPROVED** [review]
Left a few comments.

> @nemanjaglumac Can you try and run the script? See the README for setup.

Ran both the script and the tests. Both are working nicely. 🚀

File: bin/release-list/README.md [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+From this directory, run:
+
+```
+bb test:bb
+```
+
+
+## Editing the page
+
+Edit the template: [./resources/releases.md](./resources/releases-template.md).
```
Comment (jeff-bruemmer) [inline bin/release-list/README.md:31]:
```suggestion
Edit the template: [./resources/releases-template.md](./resources/releases-template.md).
```

**Review by nllho — APPROVED** [review]
Looks good :) Thank you!

**Pull request merged**